### PR TITLE
DOC: add docs for initializing rever with kinda defaultish rever.xsh

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

File: docs/usepatterns.rst [hunk 1]
```diff
@@ -21,3 +21,35 @@ there are use patterns to maximizing your productivity.
...
+    ('setup.py', 'version\s*=.*,', "version='$VERSION',")
+    ]
+    $CHANGELOG_FILENAME = 'CHANGELOG.rst'  # Filename for the changelog
+    $CHANGELOG_IGNORE = ['TEMPLATE.rst']  # Filename for the news template
+    $TAG_REMOTE = 'git@github.com:regro/rever.git'  # Repo to push tags to
+
+    $GITHUB_ORG = 'regro'  # Github org for Github releases and conda-forge
+    $GITHUB_REPO = 'rever'  # Github repo for Github releases  and conda-forge
+
+3. After setting up the ``rever.xsh`` file run ``rever setup`` in the root directory of your source repository. This will setup files and other things needed for rever to operate.
```
Comment (scopatz) [inline docs/usepatterns.rst]: Line is too long

File: docs/usepatterns.rst [hunk 2]
```diff
@@ -21,3 +21,35 @@ there are use patterns to maximizing your productivity.
...
+                  'pypi',  # Sends the package to pypi
+                  'conda_forge',  # Creates a PR into your package's feedstock
+                  'ghrelease'  # Creates a Github release entry for the new tag
+    ]
+    $VERSION_BUMP_PATTERNS = [  # These note where/how to find the version numbers
+    ('rever/__init__.py', '__version__\s*=.*', "__version__ = '$VERSION'"),
+    ('setup.py', 'version\s*=.*,', "version='$VERSION',")
+    ]
+    $CHANGELOG_FILENAME = 'CHANGELOG.rst'  # Filename for the changelog
+    $CHANGELOG_IGNORE = ['TEMPLATE.rst']  # Filename for the news template
```
Comment (scopatz) [inline docs/usepatterns.rst]: It is better to set the `$CHANGELOG_TEMPLATE` variable.

File: docs/usepatterns.rst [hunk 3]
```diff
@@ -21,3 +21,35 @@ there are use patterns to maximizing your productivity.
     # The full version, including alpha/beta/rc tags.
     release = REVER_VERSION
 
+==================
+Initializing Rever
+==================
+There are a couple steps you should take to get the most out of rever.
+1. Install rever. Rever is on conda-forge so install via ``conda install rever -c conda-forge`` or from source.
```
Comment (scopatz) [inline docs/usepatterns.rst]: rst needs blank line above numbered list

File: docs/usepatterns.rst [hunk 4]
```diff
@@ -21,3 +21,35 @@ there are use patterns to maximizing your productivity.
...
+2. Setup a ``rever.xsh`` file in the root directory of your source repository.
+Here is a simplified example from ``rever`` itself,
```
Comment (scopatz) [inline docs/usepatterns.rst]: rst needs blank line below numbered list

File: docs/usepatterns.rst [hunk 5]
```diff
@@ -21,3 +21,35 @@ there are use patterns to maximizing your productivity.
...
+                  'version_bump',  # Changes the version number in various source files (setup.py, __init__.py, etc)
+                  'changelog',  # Uses files in the news folder to create a changelog for release
+                  'tag',  # Creates a tag for the new version number
+                  'pypi',  # Sends the package to pypi
+                  'conda_forge',  # Creates a PR into your package's feedstock
+                  'ghrelease'  # Creates a Github release entry for the new tag
+    ]
+    $VERSION_BUMP_PATTERNS = [  # These note where/how to find the version numbers
+    ('rever/__init__.py', '__version__\s*=.*', "__version__ = '$VERSION'"),
+    ('setup.py', 'version\s*=.*,', "version='$VERSION',")
```
Comment (scopatz) [inline docs/usepatterns.rst]: please indent these

File: docs/usepatterns.rst (hunk 4 above)
Comment (scopatz) [reply docs/usepatterns.rst]: or this line also needs to be indented

File: docs/usepatterns.rst [hunk 6]
```diff
@@ -21,3 +21,35 @@ there are use patterns to maximizing your productivity.
...
 
+==================
+Initializing Rever
+==================
+There are a couple steps you should take to get the most out of rever.
+1. Install rever. Rever is on conda-forge so install via ``conda install rever -c conda-forge`` or from source.
+2. Setup a ``rever.xsh`` file in the root directory of your source repository.
+Here is a simplified example from ``rever`` itself,
+
+.. code-block:: xonsh
```
Comment (scopatz) [inline docs/usepatterns.rst]: This whole block needs to be indented if you want it to be under 2.

File: docs/usepatterns.rst [hunk 7]
```diff
@@ -21,3 +21,35 @@ there are use patterns to maximizing your productivity.
...
+    $CHANGELOG_FILENAME = 'CHANGELOG.rst'  # Filename for the changelog
+    $CHANGELOG_IGNORE = ['TEMPLATE.rst']  # Filename for the news template
+    $TAG_REMOTE = 'git@github.com:regro/rever.git'  # Repo to push tags to
+
+    $GITHUB_ORG = 'regro'  # Github org for Github releases and conda-forge
+    $GITHUB_REPO = 'rever'  # Github repo for Github releases  and conda-forge
+
+3. After setting up the ``rever.xsh`` file run ``rever setup`` in the root directory of your source repository. This will setup files and other things needed for rever to operate.
+4. When you are ready to release run ``rever <new_version_number>`` and rever will take care of the rest.
+5. Now you are ready, smooth sailing!
```
Comment (scopatz) [inline docs/usepatterns.rst]: This isn't a real step :)

File: docs/usepatterns.rst (hunk 3 above)
Comment (scopatz) [reply docs/usepatterns.rst]: Also, you can install rever from pypi so it should also be mentioned that you can do ``pip install re-ver``

Comment (scopatz) [reply docs/usepatterns.rst]: And this line is too long

File: rever.xsh [hunk 8]
```diff
@@ -7,7 +7,7 @@ $VERSION_BUMP_PATTERNS = [
     ('rever/__init__.py', '__version__\s*=.*', "__version__ = '$VERSION'"),
     ('setup.py', 'version\s*=.*,', "version='$VERSION',")
     ]
-$CHANGELOG_FILENAME = 'CHANGELOG.rst'
+$CHANGELOG_TEMPLATE = 'CHANGELOG.rst'
```
[the remark this answers is not in the document]
Comment (scopatz) [reply rever.xsh]: Though you can also set `$CHANGELOG_TEMPLATE = 'TEMPLATE.rst'` See https://regro.github.io/rever-docs/activities.html#changelog for more details

[the remark this answers is not in the document]
Comment (CJ-Wright) [reply rever.xsh]: sorry I put it on the wrong line.

File: docs/usepatterns.rst [hunk 9]
```diff
@@ -21,3 +21,37 @@ there are use patterns to maximizing your productivity.
...
+                      'tag',  # Creates a tag for the new version number
+                      'pypi',  # Sends the package to pypi
+                      'conda_forge',  # Creates a PR into your package's feedstock
+                      'ghrelease'  # Creates a Github release entry for the new tag
+        ]
+        $VERSION_BUMP_PATTERNS = [  # These note where/how to find the version numbers
+                                 ('rever/__init__.py', '__version__\s*=.*', "__version__ = '$VERSION'"),
+                                 ('setup.py', 'version\s*=.*,', "version='$VERSION',")
+        ]
+        $CHANGELOG_TEMPLATE = 'CHANGELOG.rst'  # Filename for the changelog
```
Comment (scopatz) [inline docs/usepatterns.rst]: should be filename

File: docs/usepatterns.rst [hunk 10]
```diff
@@ -21,3 +21,37 @@ there are use patterns to maximizing your productivity.
...
+        $CHANGELOG_IGNORE = ['TEMPLATE.rst']  # Filename for the news template
```
Comment (scopatz) [inline docs/usepatterns.rst]: should be template

File: docs/usepatterns.rst [hunk 11]
```diff
@@ -21,3 +21,39 @@ there are use patterns to maximizing your productivity.
...
     release = REVER_VERSION
 
+==================
+Initializing Rever
+==================
+There are a couple steps you should take to get the most out of rever.
+
+1. Install rever. Rever is on conda-forge so install via
+``conda install rever -c conda-forge``, via pypi with ``pip install re-ver``,
+or from source.
```
Comment (scopatz) [inline docs/usepatterns.rst]: These lines need to be indented

File: rever.xsh [hunk 12]
```diff
@@ -8,7 +8,7 @@ $VERSION_BUMP_PATTERNS = [
     ('setup.py', 'version\s*=.*,', "version='$VERSION',")
     ]
 $CHANGELOG_FILENAME = 'CHANGELOG.rst'
-$CHANGELOG_IGNORE = ['TEMPLATE.rst']
+$CHANGELOG_TEMPLATE = ['TEMPLATE.rst']
```
Comment (scopatz) [inline rever.xsh]: This envvar is a string, not a list

**Comment by scopatz** [comment]
Thanks for putting this in!

**Comment by CJ-Wright** [comment]
Sure sorry I'm failing so badly at getting the rst to behave.

**Comment by scopatz** [comment]
Looks good! Thanks!

**Pull request merged**